### ISSUE TITLE
SQLParseException[line 4:8: no viable alternative at input 'LIMIT -']

### ISSUE THREAD

**Issue by amotl** [crate/crate-python#453]
### Problem
```
sqlalchemy.exc.ProgrammingError: (crate.client.exceptions.ProgrammingError) SQLParseException[line 4:8: no viable alternative at input 'LIMIT -']
[SQL: SELECT testdrive.id, testdrive.name, testdrive.age, testdrive.gender
FROM testdrive
WHERE testdrive.name = ?
 LIMIT -1 OFFSET ?]
[parameters: ('John Doe', 0)]
(Background on this error at: https://sqlalche.me/e/14/f405)
```

### Reproduction
Enable the _Query multiple records_ code of [`cratedb-dataset-demo.py`](https://gist.github.com/amotl/eb4f6b221e81484a445bcc3417de719d) and run it.

### Reason
SQLAlchemy's [`compiler.py:limit_clause`](https://github.com/sqlalchemy/sqlalchemy/blob/rel_1_4_41/lib/sqlalchemy/sql/compiler.py#L3749-L3757) implementation:
```python
    def limit_clause(self, select, **kw):
        text = ""
        if select._limit_clause is not None:
            text += "\n LIMIT " + self.process(select._limit_clause, **kw)
        if select._offset_clause is not None:
            if select._limit_clause is None:
                text += "\n LIMIT -1"
            text += " OFFSET " + self.process(select._offset_clause, **kw)
        return text
```

**Comment by seut** [comment]
We should probably switch to extend from the `PGCompiler` instead of the default `SQLCompiler`. See https://github.com/sqlalchemy/sqlalchemy/blob/rel_1_4_41/lib/sqlalchemy/dialects/postgresql/base.py#L2394.
CrateDB (like PG) does not support negative LIMIT values.

**Comment by amotl** [comment]
Wow, thanks for quickly looking that up. My thoughts have been similar, like »let's just take the `limit_clause` from the PostgreSQL dialect«.

**Comment by amotl** [comment]
On this matter, I just discovered that `LIMIT ALL` is only possible when phrased without `OFFSET`.

Works:
```
cr> SELECT * FROM testdrive WHERE name='John Doe' limit ALL;
```

Croaks:
```
cr> SELECT * FROM testdrive WHERE name='John Doe' limit ALL OFFSET 0;
SQLParseException[Invalid LIMIT: value must be >= 0; got: -1]
```

When looking at the SQLAlchemy code, we can clearly see that it can build such an SQL statement. Please upvote if you think I should create a corresponding issue at crate/crate.

**Comment by amotl** [comment]
1. The CrateDB-related fix is https://github.com/crate/crate/pull/13108. Thanks, @matriv.
2. The SQLAlchemy-related fix is #472.
